src/main/java/nextstep/qna/service/QnAService.java

```diff
@@ -1,49 +1,52 @@
-import nextstep.qna.domain.*;
-import nextstep.users.domain.NsUser;
-import org.springframework.stereotype.Service;
-import org.springframework.transaction.annotation.Transactional;
-
-import javax.annotation.Resource;
-import java.time.LocalDateTime;
-import java.util.ArrayList;
-import java.util.List;
-
-@Service("qnaService")
-public class QnAService {
-    @Resource(name = "questionRepository")
-    private QuestionRepository questionRepository;
-
-    @Resource(name = "answerRepository")
-    private AnswerRepository answerRepository;
-
-    @Resource(name = "deleteHistoryService")
-    private DeleteHistoryService deleteHistoryService;
-
-    @Transactional
-    public void deleteQuestion(NsUser loginUser, long questionId) throws CannotDeleteException {
-        Question question = questionRepository.findById(questionId).orElseThrow(NotFoundException::new);
-        if (!question.isOwner(loginUser)) {
-            throw new CannotDeleteException("질문을 삭제할 권한이 없습니다.");
-        }
-
-        List<Answer> answers = question.getAnswers();
-        for (Answer answer : answers) {
-            if (!answer.isOwner(loginUser)) {
-                throw new CannotDeleteException("다른 사람이 쓴 답변이 있어 삭제할 수 없습니다.");
-            }
-        }
-
-        List<DeleteHistory> deleteHistories = new ArrayList<>();
-        question.setDeleted(true);
-        deleteHistories.add(new DeleteHistory(ContentType.QUESTION, questionId, question.getWriter(), LocalDateTime.now()));
-        for (Answer answer : answers) {
-            answer.setDeleted(true);
-            deleteHistories.add(new DeleteHistory(ContentType.ANSWER, answer.getId(), answer.getWriter(), LocalDateTime.now()));
-        }
-        deleteHistoryService.saveAll(deleteHistories);
-    }
-}
+package nextstep.qna.service;
+
+import java.util.List;
+import javax.annotation.Resource;
+import nextstep.qna.CannotDeleteException;
+import nextstep.qna.NotFoundException;
+import nextstep.qna.domain.*;
+import nextstep.users.domain.NsUser;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+/*
+* 깨진 코드 다시 커밋위한 주석
+* */
+
+@Service("qnaService")
+public class QnAService {
+    @Resource(name = "questionRepository")
+    private QuestionRepository questionRepository;
+    
+    @Resource(name = "answerRepository")
+    private AnswerRepository answerRepository;
+    
+    @Resource(name = "deleteHistoryService")
+    private DeleteHistoryService deleteHistoryService;
+    
+    @Transactional
+    public void deleteQuestion(NsUser loginUser, long questionId) throws CannotDeleteException {
+        Question question = questionRepository.findById(questionId).orElseThrow(NotFoundException::new);
+        question.isHaveAuthority(loginUser);
+        
+        Answers answers = question.sameUserQuestionAndAnswer();
+        
+        deleteHistoryService.saveAll(new DeleteHistories(deleteQuestion(question), deleteAnswers(answers)).toList());

```

Comment:
```suggestion
        DeleteHistories deleteHistories = question.delete(loginUser);
        deleteHistoryService.saveAll(deleteHistories);
```
Answers를 question에서 get한 후 구현할 필요가 있을까?
위와 같이 question에게 delete() 하라는 메시지를 보내는 것만으로 구현할 수 있지 않을까?

Reply:
서비스 레이어의 `deleteHistory()`는 질문을 삭제하기를 원할 뿐이지
`Answers`를 꺼내서 한다면 제3의 서비스 레이어에서 `Answers` 제거를 누락할 수있다.